docs/keycloak-install&config.md

```diff
@@ -0,0 +1,457 @@
+
+   ![Tworzenie klienta](images/keycloak-instruction-5.png)
+5. W sekcji **Valid redirect URIs**, **Valid post logout redirect URIs** oraz **Web origins** wpisujemy `*` (dla celów testowych) i klikamy **Save**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-6.png)
+
+
+---
+
+## 6. Tworzenie użytkownika
+1. Przechodzimy do **Manage → Users**.
+2. Klikamy **Create new user**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-7.png)
+3. Podajemy nazwę użytkownika, np. `zpi-user`. Opcjonalnie można dodać adres e-mail, imię i nazwisko.
+
+   ![Tworzenie użytkownika](images/keycloak-instruction-8.png)
+4. Przechodzimy do zakładki **Credentials**, klikamy **Set password**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-9.png)
+5. Ustawiamy hasło i, jeżeli nie chcemy wymuszać zmiany hasła przy pierwszym logowaniu, odznaczamy przełącznik **Temporary**.
+
+---
+
+## 7. Tworzenie ról
+### 7.1 Role Realm
+1. Przechodzimy do **Manage → Realm roles**.
+2. Klikamy **Create role**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-10.png)
+3. Podajemy nazwę roli, np. `zpi-role` i klikamy **Save**.
+
+### 7.2 Role Klienta
+1. Przechodzimy do **Manage → Clients → [nazwa klienta] → Roles**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-11.png)
+2. Klikamy **Create role**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-12.png)
+3. Podajemy nazwę roli, np. `zpi-role` i klikamy **Save**.
+
+> **Uwaga**: Nazwa roli musi odpowiadać nazwie w konfiguracji mapy uprawnień aplikacji, aby rola nadawała odpowiednie uprawnienia.
+
+---
+
+## 8. Przypisywanie ról użytkownikowi
+1. Przechodzimy do **Manage → Users → [nazwa użytkownika]**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-13.png)
+2. Wybieramy zakładkę **Role mapping** i klikamy **Assign role**.
+
+   ![Tworzenie klienta](images/keycloak-instruction-14.png)
+    - Jeżeli rola jest realmowa, wybieramy **Filter by realm roles** i przypisujemy odpowiednią rolę.
+
+      ![Tworzenie klienta](images/keycloak-instruction-15.png)
+    - Jeżeli rola jest kliencka, wybieramy **Filter by clients**, wybieramy klienta i przypisujemy rolę.
+
+   ![Tworzenie klienta](images/keycloak-instruction-16.png)
+
+> Można użyć pola wyszukiwania obok filtrów, aby szybko znaleźć rolę.
+
+---
+
+## 9. Konfiguracja zmiennych środowiskowych
+Zmienne środowiskowe niezbęne do zaimportowania w związku z Keycloackiem to:
+```env
+VITE_KEYCLOAK_URL=http://{adres-keycloak}:{port-keycloak}/
+VITE_KEYCLOAK_CLIENTNAME={nazwa-clienta}
+VITE_KEYCLOAK_REALMNAME={nazwa-realm}
+```
+W przypadku przykładowej konfiguracji będzie to:
+```env
+VITE_KEYCLOAK_URL=http://localhost:4000/
+VITE_KEYCLOAK_CLIENTNAME=ZPI-client
+VITE_KEYCLOAK_REALMNAME=ZPI-realm
+```
+
+Korzystając z plików `.env` trzeba dodać je dodać w dwóch różnych miejscach zarówno dla backendu jak i frontendu:
+##### Backend (`KuberbetesUserManager/backend/.env`)
+##### Frontend (`KuberbetesUserManager/frontend/.env`)
```

Comment:
Skoro mamy osobne pliki env do backendu i frontendu, to czy jest sens dawać VITE_ na backendzie?

Reply:
Tak, bo docelowo się raczej korzysta ze zmjeinnycb środowiskowych a nie plików .env i warto wtedy ujednolicić te zmienne, żeby nie trzeba było ustawiać ich dwukrotnie

Reply:
ok